In [50]:
import pymysql
import pandas as pd
'''
def init_connection():
    connection = pymysql.connect(
        host="localhost",
        port=3306,
        user="root",
        password="root",
        cursorclass=pymysql.cursors.DictCursor,
        charset="utf8",
    )
    return connection

sql = "SHOW DATABASES;"
conn = init_connection()

with conn:
    with conn.cursor() as cur:
        cur.execute(sql)
        for data in cur:
            print(data)

'''

'\ndef init_connection():\n    connection = pymysql.connect(\n        host="localhost",\n        port=3306,\n        user="root",\n        password="root",\n        cursorclass=pymysql.cursors.DictCursor,\n        charset="utf8",\n    )\n    return connection\n\nsql = "SHOW DATABASES;"\nconn = init_connection()\n\nwith conn:\n    with conn.cursor() as cur:\n        cur.execute(sql)\n        for data in cur:\n            print(data)\n\n'

In [19]:
'''
conn.close()
'''

'\nconn.close()\n'

In [15]:
'''
conn = init_connection()
sql = "CREATE DATABASE address DEFAULT CHARACTER SET utf8;"

with conn:
    with conn.cursor() as cur:
        cur.execute(sql)
        conn.commit()
'''

'\nconn = init_connection()\nsql = "CREATE DATABASE address DEFAULT CHARACTER SET utf8;"\n\nwith conn:\n    with conn.cursor() as cur:\n        cur.execute(sql)\n        conn.commit()\n'

In [17]:
conn = init_connection()
sql = "SHOW DATABASES;"

with conn:
    with conn.cursor() as cur:
        cur.execute(sql)
        for data in cur:
            print(data)

{'Database': 'address'}
{'Database': 'information_schema'}
{'Database': 'mysql'}
{'Database': 'performance_schema'}
{'Database': 'sys'}


# 5.4 - 테이블 생성

In [20]:
def init_db_connection():
    connection = pymysql.connect(
        host="localhost",
        port=3306,
        user="root",
        password="root",
        database="address",
        cursorclass=pymysql.cursors.DictCursor,
        charset="utf8",
        local_infile=True
    )
    return connection

# 디폴트 접속 데이터베스 - adress

In [21]:
def query_get(sql):
    connection = init_db_connection() 
    with connection: # 구문이 끝나고 연결을 닫아줌
        with connection.cursor() as cursor:  # 객체 생성, 관리
            cursor.execute(sql) # sql 쿼리 실행
            return cursor.fetchall() # 메소드 커서가 가지고있는 결과 집합에서 모든 행을 가져옴


# 데이터베이스(adress)에 연결하여 SQL 쿼리를 실행하고, 그 결과를 모두 가져와 반환

In [23]:
def query_update(sql):
    connection = init_db_connection()
    with connection:
        with connection.cursor() as cursor:
            cursor.execute(sql)
            connection.commit() #데이터베이스에 대한 모든 변경 사항을 커밋함
            return True # 잘 실행되었다면 True 반환
        
# sql에 담긴 쿼리를 실행하고 잘 실행되었다면 True를 반환
# 쿼리에 담긴 질의가 CREATE와 같이 데이터를 업로드하거나 조작할 때 사용

In [24]:
# 테이블 생성

sql = '''
    CREATE TABLE `rnaddrkor` (
        `도로명주소관리번호` varchar(26) NOT NULL,
        `법정동코드` varchar(10),
        `시도명` varchar(40),
        `시군구명` varchar(40),
        `읍면동명` varchar(40),
        `리명` varchar(40),
        `산여부` varchar(1),
        `번지` varchar(4),
        `호` varchar(4),
        `도로명코드` varchar(12) NOT NULL,
        `도로명` varchar(80),
        `지하여부` varchar(1) NOT NULL,
        `건물본번` int(5) NOT NULL,
        `건물부번` int(5) NOT NULL,
        `행정동코드` varchar(60),
        `행정동명` varchar(60),
        `기초구역번호(우편번호)` varchar(5),
        `이전도로명주소` varchar(400),
        `효력발생일` varchar(8),
        `공동주택구분` varchar(1),
        `이동사유코드` varchar(2),
        `건축물대장건물명` varchar(400),
        `시군구용건물명` varchar(400),
        `비고` varchar(200),
        PRIMARY KEY (`도로명주소관리번호`, `도로명코드`, `지하여부`, `건물본번`, `건물부번`)
    );
'''

query_update(sql)

True

In [25]:
# 테이블 생성 2
sql = '''
    CREATE TABLE `jibun_rnaddrkor` (
        `도로명주소관리번호` varchar(26) NOT NULL,
        `법정동코드` varchar(10) NOT NULL,
        `시도명` varchar(40),
        `시군구명` varchar(40),
        `법정읍면동명` varchar(40),
        `법정리명` varchar(40),
        `산여부` varchar(1) NOT NULL,
        `지번본번(번지)` int(4) NOT NULL,
        `지번부번(호)` int(4) NOT NULL,
        `도로명코드` varchar(12),
        `지하여부` varchar(1),
        `건물본번` int(5),
        `건물부번` int(5),
        `이동사유코드` varchar(2),
        PRIMARY KEY (`도로명주소관리번호`, `법정동코드`, `산여부`, `지번본번(번지)`, `지번부번(호)`)
    );
'''

query_update(sql)

True

In [26]:
# 테이블 이름 불러오기
query_get("SHOW TABLES;")

[{'Tables_in_address': 'jibun_rnaddrkor'}, {'Tables_in_address': 'rnaddrkor'}]

In [27]:
#  rnaddrkor의 스키마 확인
query_get("DESC rnaddrkor;")


[{'Field': '도로명주소관리번호',
  'Type': 'varchar(26)',
  'Null': 'NO',
  'Key': 'PRI',
  'Default': None,
  'Extra': ''},
 {'Field': '법정동코드',
  'Type': 'varchar(10)',
  'Null': 'YES',
  'Key': '',
  'Default': None,
  'Extra': ''},
 {'Field': '시도명',
  'Type': 'varchar(40)',
  'Null': 'YES',
  'Key': '',
  'Default': None,
  'Extra': ''},
 {'Field': '시군구명',
  'Type': 'varchar(40)',
  'Null': 'YES',
  'Key': '',
  'Default': None,
  'Extra': ''},
 {'Field': '읍면동명',
  'Type': 'varchar(40)',
  'Null': 'YES',
  'Key': '',
  'Default': None,
  'Extra': ''},
 {'Field': '리명',
  'Type': 'varchar(40)',
  'Null': 'YES',
  'Key': '',
  'Default': None,
  'Extra': ''},
 {'Field': '산여부',
  'Type': 'varchar(1)',
  'Null': 'YES',
  'Key': '',
  'Default': None,
  'Extra': ''},
 {'Field': '번지',
  'Type': 'varchar(4)',
  'Null': 'YES',
  'Key': '',
  'Default': None,
  'Extra': ''},
 {'Field': '호',
  'Type': 'varchar(4)',
  'Null': 'YES',
  'Key': '',
  'Default': None,
  'Extra': ''},
 {'Field': '도로명코드',
  '

# 5.5 - 데이터 삽입

In [28]:
'''
def init_db_connection():
    connection = pymysql.connect(
        host="localhost",
        port=3306,
        user="root",
        password="root",
        database="address",
        cursorclass=pymysql.cursors.DictCursor,
        charset="utf8",
        local_infile=True # 클라이언스 상에서 조건 추가
    )
    return connection
'''

# 5.4에서 이미 정의된 함수

'\ndef init_db_connection():\n    connection = pymysql.connect(\n        host="localhost",\n        port=3306,\n        user="root",\n        password="root",\n        database="address",\n        cursorclass=pymysql.cursors.DictCursor,\n        charset="utf8",\n        local_infile=True # 클라이언스 상에서 조건 추가\n    )\n    return connection\n'

In [31]:
import glob
file_list = glob.glob("week3_data/rnaddrkor/rnaddrkor/rnaddrkor_*.txt")

for file in file_list:
    print(file)

week3_data/rnaddrkor/rnaddrkor\rnaddrkor_busan.txt
week3_data/rnaddrkor/rnaddrkor\rnaddrkor_chungbuk.txt
week3_data/rnaddrkor/rnaddrkor\rnaddrkor_chungnam.txt
week3_data/rnaddrkor/rnaddrkor\rnaddrkor_daegu.txt
week3_data/rnaddrkor/rnaddrkor\rnaddrkor_daejeon.txt
week3_data/rnaddrkor/rnaddrkor\rnaddrkor_gangwon.txt
week3_data/rnaddrkor/rnaddrkor\rnaddrkor_gwangju.txt
week3_data/rnaddrkor/rnaddrkor\rnaddrkor_gyeongbuk.txt
week3_data/rnaddrkor/rnaddrkor\rnaddrkor_gyeongnam.txt
week3_data/rnaddrkor/rnaddrkor\rnaddrkor_gyunggi.txt
week3_data/rnaddrkor/rnaddrkor\rnaddrkor_incheon.txt
week3_data/rnaddrkor/rnaddrkor\rnaddrkor_jeju.txt
week3_data/rnaddrkor/rnaddrkor\rnaddrkor_jeonbuk.txt
week3_data/rnaddrkor/rnaddrkor\rnaddrkor_jeonnam.txt
week3_data/rnaddrkor/rnaddrkor\rnaddrkor_sejong.txt
week3_data/rnaddrkor/rnaddrkor\rnaddrkor_seoul.txt
week3_data/rnaddrkor/rnaddrkor\rnaddrkor_ulsan.txt


In [32]:
import os
from tqdm import tqdm

total_line_count = 0  # 총 행수를 저장할 변수

for file in tqdm(file_list, desc='Processing files'):
    file_path = os.path.abspath(file)

    # cp949로 파일 읽고 행 불러오기
    with open(file_path, 'r', encoding='cp949', errors='ignore') as f:
        lines = f.readlines()
    
    # 행 수 계산
    line_count = len(lines)
    
    # 총 행 수에 더하기
    total_line_count += line_count

    # cp949로 읽은 파일을 utf8로 임시파일에 저장
    temp_file_path = "temp_file.txt"
    with open(temp_file_path, 'w', encoding='utf8') as f:
        f.writelines(lines)

    # rnaddrkor에 데이터 로드 , 구분자 |
    sql = f'''
        LOAD DATA LOCAL INFILE "{temp_file_path}" INTO TABLE rnaddrkor
        FIELDS TERMINATED BY "|";
    '''

    # 해당 파일 경로 / 행수
    print(f"Processing file: {file_path}, Number of lines: {line_count}")
    query_update(sql)

    # 임시파일 삭제
    os.remove(temp_file_path)

# 총 행수
print(f"Total number of lines: {total_line_count}")

Processing files:   0%|          | 0/17 [00:00<?, ?it/s]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\rnaddrkor_busan.txt, Number of lines: 301822


Processing files:   6%|▌         | 1/17 [00:03<01:01,  3.84s/it]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\rnaddrkor_chungbuk.txt, Number of lines: 335225


Processing files:  12%|█▏        | 2/17 [00:07<00:58,  3.92s/it]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\rnaddrkor_chungnam.txt, Number of lines: 491608


Processing files:  18%|█▊        | 3/17 [00:13<01:06,  4.78s/it]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\rnaddrkor_daegu.txt, Number of lines: 229727


Processing files:  24%|██▎       | 4/17 [00:16<00:51,  3.98s/it]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\rnaddrkor_daejeon.txt, Number of lines: 114026


Processing files:  29%|██▉       | 5/17 [00:17<00:36,  3.02s/it]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\rnaddrkor_gangwon.txt, Number of lines: 364686


Processing files:  35%|███▌      | 6/17 [00:22<00:38,  3.53s/it]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\rnaddrkor_gwangju.txt, Number of lines: 120474


Processing files:  41%|████      | 7/17 [00:23<00:28,  2.83s/it]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\rnaddrkor_gyeongbuk.txt, Number of lines: 713862


Processing files:  47%|████▋     | 8/17 [00:32<00:41,  4.65s/it]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\rnaddrkor_gyeongnam.txt, Number of lines: 653480


Processing files:  53%|█████▎    | 9/17 [00:40<00:45,  5.71s/it]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\rnaddrkor_gyunggi.txt, Number of lines: 1020563


Processing files:  59%|█████▉    | 10/17 [00:52<00:54,  7.75s/it]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\rnaddrkor_incheon.txt, Number of lines: 186039


Processing files:  65%|██████▍   | 11/17 [00:54<00:36,  6.13s/it]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\rnaddrkor_jeju.txt, Number of lines: 155426


Processing files:  71%|███████   | 12/17 [00:56<00:24,  4.85s/it]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\rnaddrkor_jeonbuk.txt, Number of lines: 437063


Processing files:  76%|███████▋  | 13/17 [01:02<00:20,  5.04s/it]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\rnaddrkor_jeonnam.txt, Number of lines: 598298


Processing files:  82%|████████▏ | 14/17 [01:09<00:17,  5.77s/it]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\rnaddrkor_sejong.txt, Number of lines: 27510


Processing files:  88%|████████▊ | 15/17 [01:10<00:08,  4.16s/it]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\rnaddrkor_seoul.txt, Number of lines: 530721


Processing files:  94%|█████████▍| 16/17 [01:16<00:04,  4.83s/it]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\rnaddrkor_ulsan.txt, Number of lines: 104458


Processing files: 100%|██████████| 17/17 [01:17<00:00,  4.59s/it]

Total number of lines: 6384988


In [34]:
# 이미 지운거 아닌가...?

'''
LOAD DATA LOCAL INFILE "{temp_file_path}" INTO TABLE rnaddrkor \\
    FIELDS TERMINATED BY "|";
'''

'\nLOAD DATA LOCAL INFILE "{temp_file_path}" INTO TABLE rnaddrkor \\\n    FIELDS TERMINATED BY "|";\n'

In [37]:
sql = "SELECT * FROM rnaddrkor LIMIT 5;"
'''
sql = "SELECT * FROM rnaddrkor LIMIT 5;" 이 구문은 SQL(Structured Query Language) 쿼리의 한 형태로, 
특정 데이터베이스 내의 rnaddrkor라는 이름의 테이블에서 모든 컬럼(*로 표현)의 데이터를 조회하는 명령입니다. 
여기서 LIMIT 5는 조회 결과의 수를 처음 5개 행으로 제한하라는 의미입니다. 간단히 말해서, 
이 쿼리는 rnaddrkor 테이블의 상위 5개 행에 있는 모든 데이터를 선택하라는 의미입니다.

(by gpt)
'''

query_get(sql)

[{'도로명주소관리번호': '11110101310001200009400000',
  '법정동코드': '1111010100',
  '시도명': '서울특별시',
  '시군구명': '종로구',
  '읍면동명': '청운동',
  '리명': '',
  '산여부': '0',
  '번지': '144',
  '호': '3',
  '도로명코드': '111103100012',
  '도로명': '자하문로',
  '지하여부': '0',
  '건물본번': 94,
  '건물부번': 0,
  '행정동코드': '1111051500',
  '행정동명': '청운효자동',
  '기초구역번호(우편번호)': '03047',
  '이전도로명주소': '',
  '효력발생일': '20110729',
  '공동주택구분': '0',
  '이동사유코드': '',
  '건축물대장건물명': '',
  '시군구용건물명': '',
  '비고': '\r'},
 {'도로명주소관리번호': '11110101310001200009600000',
  '법정동코드': '1111010100',
  '시도명': '서울특별시',
  '시군구명': '종로구',
  '읍면동명': '청운동',
  '리명': '',
  '산여부': '0',
  '번지': '108',
  '호': '14',
  '도로명코드': '111103100012',
  '도로명': '자하문로',
  '지하여부': '0',
  '건물본번': 96,
  '건물부번': 0,
  '행정동코드': '1111051500',
  '행정동명': '청운효자동',
  '기초구역번호(우편번호)': '03047',
  '이전도로명주소': '',
  '효력발생일': '20110729',
  '공동주택구분': '1',
  '이동사유코드': '',
  '건축물대장건물명': '',
  '시군구용건물명': '평안빌',
  '비고': '\r'},
 {'도로명주소관리번호': '11110101310001200009800000',
  '법정동코드': '1111010100',
  '시도명': '서울특별시'

In [38]:
sql = "SELECT COUNT(*) FROM rnaddrkor;"

'''
SELECT COUNT(*): SELECT 명령어는 데이터를 조회할 때 사용되며, COUNT(*)는 특정 조건에 맞는 행의 개수를 세라는 의미입니다. 
여기서 *는 모든 행을 의미하므로, COUNT(*)는 테이블의 모든 행을 대상으로 개수를 세라는 것을 나타냅니다.

(by gpt)
'''
query_get(sql)

[{'COUNT(*)': 6384988}]

In [40]:
# 지번주소도 같은 코드 실행

import glob

file_list = glob.glob("week3_data/rnaddrkor/rnaddrkor/jibun_*.txt")

for file in file_list:
    print(file)

total_line_count = 0  # 총 행수를 저장할 변수

for file in tqdm(file_list, desc='Processing files'):
    file_path = os.path.abspath(file)

    # cp949로 파일 읽고 행 불러오기
    with open(file_path, 'r', encoding='cp949', errors='ignore') as f:
        lines = f.readlines()
    
    # 행 수 계산
    line_count = len(lines)
    
    # 총 행 수에 더하기
    total_line_count += line_count

    # cp949로 읽은 파일을 utf8로 임시파일에 저장
    temp_file_path = "temp_file.txt"
    with open(temp_file_path, 'w', encoding='utf8') as f:
        f.writelines(lines)

    sql = f'''
        LOAD DATA LOCAL INFILE "{temp_file_path}" INTO TABLE jibun_rnaddrkor
        FIELDS TERMINATED BY "|";
    '''
    print(f"Processing file: {file_path}, Number of lines: {line_count}")
    query_update(sql)

    # 임시파일 삭제
    os.remove(temp_file_path)

print(f"Total number of lines: {total_line_count}")

week3_data/rnaddrkor/rnaddrkor\jibun_rnaddrkor_busan.txt
week3_data/rnaddrkor/rnaddrkor\jibun_rnaddrkor_chungbuk.txt
week3_data/rnaddrkor/rnaddrkor\jibun_rnaddrkor_chungnam.txt
week3_data/rnaddrkor/rnaddrkor\jibun_rnaddrkor_daegu.txt
week3_data/rnaddrkor/rnaddrkor\jibun_rnaddrkor_daejeon.txt
week3_data/rnaddrkor/rnaddrkor\jibun_rnaddrkor_gangwon.txt
week3_data/rnaddrkor/rnaddrkor\jibun_rnaddrkor_gwangju.txt
week3_data/rnaddrkor/rnaddrkor\jibun_rnaddrkor_gyeongbuk.txt
week3_data/rnaddrkor/rnaddrkor\jibun_rnaddrkor_gyeongnam.txt
week3_data/rnaddrkor/rnaddrkor\jibun_rnaddrkor_gyunggi.txt
week3_data/rnaddrkor/rnaddrkor\jibun_rnaddrkor_incheon.txt
week3_data/rnaddrkor/rnaddrkor\jibun_rnaddrkor_jeju.txt
week3_data/rnaddrkor/rnaddrkor\jibun_rnaddrkor_jeonbuk.txt
week3_data/rnaddrkor/rnaddrkor\jibun_rnaddrkor_jeonnam.txt
week3_data/rnaddrkor/rnaddrkor\jibun_rnaddrkor_sejong.txt
week3_data/rnaddrkor/rnaddrkor\jibun_rnaddrkor_seoul.txt
week3_data/rnaddrkor/rnaddrkor\jibun_rnaddrkor_ulsan.txt


Processing files:   0%|          | 0/17 [00:00<?, ?it/s]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\jibun_rnaddrkor_busan.txt, Number of lines: 42193


Processing files:   6%|▌         | 1/17 [00:00<00:07,  2.18it/s]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\jibun_rnaddrkor_chungbuk.txt, Number of lines: 84792


Processing files:  12%|█▏        | 2/17 [00:01<00:10,  1.47it/s]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\jibun_rnaddrkor_chungnam.txt, Number of lines: 186499


Processing files:  18%|█▊        | 3/17 [00:03<00:16,  1.17s/it]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\jibun_rnaddrkor_daegu.txt, Number of lines: 28049


Processing files:  29%|██▉       | 5/17 [00:03<00:06,  1.73it/s]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\jibun_rnaddrkor_daejeon.txt, Number of lines: 11926
Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\jibun_rnaddrkor_gangwon.txt, Number of lines: 178859


Processing files:  41%|████      | 7/17 [00:05<00:07,  1.39it/s]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\jibun_rnaddrkor_gwangju.txt, Number of lines: 17851
Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\jibun_rnaddrkor_gyeongbuk.txt, Number of lines: 143788


Processing files:  47%|████▋     | 8/17 [00:06<00:08,  1.11it/s]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\jibun_rnaddrkor_gyeongnam.txt, Number of lines: 300283


Processing files:  53%|█████▎    | 9/17 [00:09<00:12,  1.50s/it]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\jibun_rnaddrkor_gyunggi.txt, Number of lines: 285178


Processing files:  59%|█████▉    | 10/17 [00:12<00:13,  1.88s/it]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\jibun_rnaddrkor_incheon.txt, Number of lines: 31745


Processing files:  71%|███████   | 12/17 [00:12<00:05,  1.04s/it]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\jibun_rnaddrkor_jeju.txt, Number of lines: 19240
Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\jibun_rnaddrkor_jeonbuk.txt, Number of lines: 114937


Processing files:  76%|███████▋  | 13/17 [00:13<00:04,  1.07s/it]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\jibun_rnaddrkor_jeonnam.txt, Number of lines: 308309


Processing files:  82%|████████▏ | 14/17 [00:16<00:04,  1.66s/it]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\jibun_rnaddrkor_sejong.txt, Number of lines: 5327
Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\jibun_rnaddrkor_seoul.txt, Number of lines: 86986


Processing files: 100%|██████████| 17/17 [00:18<00:00,  1.14it/s]

Processing file: c:\Users\82102\Desktop\팀케일리 스터디\week3_data\rnaddrkor\rnaddrkor\jibun_rnaddrkor_ulsan.txt, Number of lines: 18629


Processing files: 100%|██████████| 17/17 [00:18<00:00,  1.06s/it]

Total number of lines: 1864591


In [41]:
sql = "SELECT * FROM jibun_rnaddrkor LIMIT 5;"
query_get(sql)

[{'도로명주소관리번호': '11110101310001200009900004',
  '법정동코드': '1111010100',
  '시도명': '서울특별시',
  '시군구명': '종로구',
  '법정읍면동명': '청운동',
  '법정리명': '',
  '산여부': '0',
  '지번본번(번지)': 130,
  '지번부번(호)': 3,
  '도로명코드': '111103100012',
  '지하여부': '0',
  '건물본번': 99,
  '건물부번': 4,
  '이동사유코드': '\r'},
 {'도로명주소관리번호': '11110101310001200010100000',
  '법정동코드': '1111010100',
  '시도명': '서울특별시',
  '시군구명': '종로구',
  '법정읍면동명': '청운동',
  '법정리명': '',
  '산여부': '0',
  '지번본번(번지)': 129,
  '지번부번(호)': 2,
  '도로명코드': '111103100012',
  '지하여부': '0',
  '건물본번': 101,
  '건물부번': 0,
  '이동사유코드': '\r'},
 {'도로명주소관리번호': '11110101310001200010100000',
  '법정동코드': '1111010100',
  '시도명': '서울특별시',
  '시군구명': '종로구',
  '법정읍면동명': '청운동',
  '법정리명': '',
  '산여부': '0',
  '지번본번(번지)': 129,
  '지번부번(호)': 3,
  '도로명코드': '111103100012',
  '지하여부': '0',
  '건물본번': 101,
  '건물부번': 0,
  '이동사유코드': '\r'},
 {'도로명주소관리번호': '11110101310001200010100000',
  '법정동코드': '1111010100',
  '시도명': '서울특별시',
  '시군구명': '종로구',
  '법정읍면동명': '청운동',
  '법정리명': '',
  '산여부': '0',
  '지번본번(번지)': 131,
  

In [42]:
sql = "SELECT COUNT(*) FROM jibun_rnaddrkor;"
query_get(sql)

[{'COUNT(*)': 1864591}]

# 5.6 - SQL을 통한 데이터 분석

CREATE TABLE full_rna_addr AS 새로운 테이블 full_rna_addr를 생성하고, 특정 조건 (뒤에 나오는 것들)을 만족하는 데이터를 삽입   
SELECT DISTINCT - 중복 제거   
CASE -- WHEN -- THEN -- ELSE -- END    
   
CASE 컬럼명 | 표현식 WHEN 조건식1 THEN 결과1   
        WHEN 조건식2 THEN 결과2   
        .......   
        WHEN 조건식n THEN 결과n   
        ELSE 결과   
END   
   
형태로 사용   

* 예시 

case when x = y then a else b end   
: 조건 x=y 가 true 일 경우 a 이고 그렇지 않으면 b   
case when x < y then a when x = y then b else c end   
: 조건 x<y 가 true 일 경우 a 로, 조건 x = y 일 경우엔 b 로 그렇지 않으면 c 로 변경   
case XYZ when 'foo' then 'moo' else 'bar' end   
: XYZ 가 foo 일 경우 moo 로 변경, 그렇지 않으면 bar 로 변경   

In [44]:
sql = '''
CREATE TABLE full_rna_addr AS
SELECT DISTINCT A.`기초구역번호(우편번호)` AS 새우편번호,
    CONCAT(A.시도명, ' ',
        CASE WHEN A.시군구명 = '' THEN '' ELSE CONCAT(A.시군구명, ' ') END,
        CASE WHEN A.읍면동명 = '' THEN '' ELSE
            CASE WHEN A.리명 = '' THEN '' ELSE CONCAT(A.읍면동명, ' ') END
        END,
        A.도로명, ' ',
        CASE WHEN A.지하여부 = 0 THEN ''
            WHEN A.지하여부 = 1 THEN '지하 '
            WHEN A.지하여부 = 2 THEN '공중 '
            ELSE ''
        END,
        A.건물본번,
        CASE WHEN A.건물부번 = 0 THEN '' ELSE CONCAT('-', A.건물부번) END,
        CASE
            WHEN A.공동주택구분 = '0' THEN
                CASE
                    WHEN A.읍면동명 = '' THEN ''
                    ELSE
                        CASE WHEN A.리명 = '' THEN CONCAT(' (', A.읍면동명, ')') ELSE '' END
                END
            WHEN A.공동주택구분 = '1' THEN
                CASE
                    WHEN A.읍면동명 = '' THEN
                        CASE
                            WHEN A.시군구용건물명 = '' THEN ''
                            ELSE CONCAT(' (', A.시군구용건물명, ')')
                        END
                    ELSE CONCAT(' (',
                        CASE
                            WHEN A.리명 = '' THEN CONCAT(A.읍면동명, ', ')
                            ELSE ''
                        END,
                        CASE
                            WHEN A.시군구용건물명 = '' THEN ''
                            ELSE A.시군구용건물명
                        END,
                        ')')
                END
            ELSE ''
        END
    ) AS 도로명주소1,
    CONCAT(A.시도명, ' ',
        CASE WHEN A.시군구명 = '' THEN '' ELSE CONCAT(A.시군구명, ' ') END,
        CASE WHEN A.읍면동명 = '' THEN '' ELSE
            CASE WHEN A.리명 = '' THEN '' ELSE CONCAT(A.읍면동명, ' ') END
        END,
        A.도로명, ' ',
        CASE WHEN A.지하여부 = 0 THEN ''
            WHEN A.지하여부 = 1 THEN '지하 '
            WHEN A.지하여부 = 2 THEN '공중 '
            ELSE ''
        END,
        A.건물본번,
        CASE WHEN A.건물부번 = 0 THEN '' ELSE CONCAT('-', A.건물부번) END
    ) AS 도로명주소2,
    A.도로명주소관리번호 AS 도로명주소관리번호,
    A.시도명 AS 시도명
FROM rnaddrkor A;
'''

# 새로운 테이블 full_rna_addr를 생성하고, 기존 rnaddrkor 테이블에서 특정 조건으로 데이터를 선택하여 삽입
# SELECT DISTINCT - 중복 제거
# CASE -- WHEN -- THEN -- ELSE -- END 
""" 
CASE 컬럼명 | 표현식 WHEN 조건식1 THEN 결과1
    WHEN 조건식2 THEN 결과2
    .......
    WHEN 조건식n THEN 결과n
    ELSE 결과
END

형태로 사용

- 예시 -
case when x = y then a else b end 
: 조건 x=y 가 true 일 경우 a 이고 그렇지 않으면 b
case when x < y then a when x = y then b else c end
: 조건 x<y 가 true 일 경우 a 로, 조건 x = y 일 경우엔 b 로 그렇지 않으면 c 로 변경
case XYZ when 'foo' then 'moo' else 'bar' end
: XYZ 가 foo 일 경우 moo 로 변경, 그렇지 않으면 bar 로 변경
"""
query_update(sql)

True

In [45]:
# 위에 설명 있음
sql = "SELECT * FROM full_rna_addr LIMIT 10;"

query_get(sql)

[{'새우편번호': '03047',
  '도로명주소1': '서울특별시 종로구 자하문로 94 (청운동)',
  '도로명주소2': '서울특별시 종로구 자하문로 94',
  '도로명주소관리번호': '11110101310001200009400000',
  '시도명': '서울특별시'},
 {'새우편번호': '03047',
  '도로명주소1': '서울특별시 종로구 자하문로 96 (청운동, 평안빌)',
  '도로명주소2': '서울특별시 종로구 자하문로 96',
  '도로명주소관리번호': '11110101310001200009600000',
  '시도명': '서울특별시'},
 {'새우편번호': '03047',
  '도로명주소1': '서울특별시 종로구 자하문로 98 (청운동, 청운빌라)',
  '도로명주소2': '서울특별시 종로구 자하문로 98',
  '도로명주소관리번호': '11110101310001200009800000',
  '시도명': '서울특별시'},
 {'새우편번호': '03032',
  '도로명주소1': '서울특별시 종로구 자하문로 99-3 (청운동, 풍림팍사이드빌라)',
  '도로명주소2': '서울특별시 종로구 자하문로 99-3',
  '도로명주소관리번호': '11110101310001200009900003',
  '시도명': '서울특별시'},
 {'새우편번호': '03032',
  '도로명주소1': '서울특별시 종로구 자하문로 99-4 (청운동)',
  '도로명주소2': '서울특별시 종로구 자하문로 99-4',
  '도로명주소관리번호': '11110101310001200009900004',
  '시도명': '서울특별시'},
 {'새우편번호': '03047',
  '도로명주소1': '서울특별시 종로구 자하문로 100 (청운동)',
  '도로명주소2': '서울특별시 종로구 자하문로 100',
  '도로명주소관리번호': '11110101310001200010000000',
  '시도명': '서울특별시'},
 {'새우편번호': '03047',
  '도로명주소1': '

* SELECT `시도명`, COUNT(`도로명주소관리번호`) AS count FROM full_rna_addr   
full_rna_addr에서 시도명, 도로명주소 관리번호 갯수를 선택

* GROUP BY `시도명`   
시도명으로 묶기

* ORDER BY count DESC   
결과값 내림차순으로 정렬

In [47]:
sql = "SELECT `시도명`, COUNT(`도로명주소관리번호`) AS count FROM full_rna_addr GROUP BY `시도명` ORDER BY count DESC;"
query_get(sql)


[{'시도명': '경기도', 'count': 1020563},
 {'시도명': '경상북도', 'count': 713862},
 {'시도명': '경상남도', 'count': 653480},
 {'시도명': '전라남도', 'count': 598298},
 {'시도명': '서울특별시', 'count': 530721},
 {'시도명': '충청남도', 'count': 491608},
 {'시도명': '전북특별자치도', 'count': 437063},
 {'시도명': '강원특별자치도', 'count': 364686},
 {'시도명': '충청북도', 'count': 335225},
 {'시도명': '부산광역시', 'count': 301822},
 {'시도명': '대구광역시', 'count': 229727},
 {'시도명': '인천광역시', 'count': 186039},
 {'시도명': '제주특별자치도', 'count': 155426},
 {'시도명': '광주광역시', 'count': 120474},
 {'시도명': '대전광역시', 'count': 114026},
 {'시도명': '울산광역시', 'count': 104458},
 {'시도명': '세종특별자치시', 'count': 27510}]

engine.connect(): 앞서 생성한 엔진을 사용하여 데이터베이스에 연결하고, 연결 객체를 반환   
pd.read_sql(sql, con=conn): Pandas의 read_sql 함수를 사용하여 SQL 쿼리를 실행하고, 그 결과를 Pandas DataFrame(df)에 로드   
conn.close(): 데이터베이스 연결을 안전하게 종료

In [51]:
from sqlalchemy import create_engine

engine = create_engine("mysql+pymysql://root:root@localhost:3306/address?charset=utf8", echo=True)

conn = engine.connect()
df = pd.read_sql(sql, con=conn)
conn.close()

df

2024-05-24 01:33:16,250 INFO sqlalchemy.engine.Engine SELECT DATABASE()
2024-05-24 01:33:16,251 INFO sqlalchemy.engine.Engine [raw sql] {}
2024-05-24 01:33:16,253 INFO sqlalchemy.engine.Engine SELECT @@sql_mode
2024-05-24 01:33:16,253 INFO sqlalchemy.engine.Engine [raw sql] {}
2024-05-24 01:33:16,254 INFO sqlalchemy.engine.Engine SELECT @@lower_case_table_names
2024-05-24 01:33:16,255 INFO sqlalchemy.engine.Engine [raw sql] {}
2024-05-24 01:33:16,257 INFO sqlalchemy.engine.Engine BEGIN (implicit)
2024-05-24 01:33:16,257 INFO sqlalchemy.engine.Engine DESCRIBE `address`.`SELECT ``시도명``, COUNT(``도로명주소관리번호``) AS count FROM full_rna_addr GROUP BY ``시도명`` ORDER BY count DESC;`
2024-05-24 01:33:16,258 INFO sqlalchemy.engine.Engine [raw sql] {}
2024-05-24 01:33:16,259 INFO sqlalchemy.engine.Engine SELECT `시도명`, COUNT(`도로명주소관리번호`) AS count FROM full_rna_addr GROUP BY `시도명` ORDER BY count DESC;
2024-05-24 01:33:16,260 INFO sqlalchemy.engine.Engine [raw sql] {}
2024-05-24 01:33:20,994 INFO sqlalc

시도명    count
0       경기도  1020563
1      경상북도   713862
2      경상남도   653480
3      전라남도   598298
4     서울특별시   530721
5      충청남도   491608
6   전북특별자치도   437063
7   강원특별자치도   364686
8      충청북도   335225
9     부산광역시   301822
10    대구광역시   229727
11    인천광역시   186039
12  제주특별자치도   155426
13    광주광역시   120474
14    대전광역시   114026
15    울산광역시   104458
16  세종특별자치시    27510

In [60]:
import plotly.graph_objects as go
import pandas as pd

# 바그래프 생성
bar = go.Bar(x=df['시도명'], y=df['count'], name='도로명주소관리번호 개수')

# 레이아웃 설정
layout = go.Layout(title='시도별 도로명주소의 개수')

# Figure 생성과 출력
fig = go.Figure(data=bar, layout=layout)
fig.show()

# 지번 - 관련 지번 많은순 분석
#### 지번 테이블에서 PNU 생성

PNU = 필지 고유번호   
시도(2) - 시군구(3) - 읍면동(3) - 리(2) - 필지구분(1) - 본번(4) - 부번(4)   
총 19자

In [53]:
# 여러 쿼리를 실행하는 함수

def multi_query_update(sql):
    connection = init_db_connection() # 데이터베이스 연결 초기화
    with connection: 
        with connection.cursor() as cursor: # 블록 끝나면 커서 종료
            sql_list = [query.strip() for query in sql.strip().split('\n')]
            # 무슨 의미인지 모르겠습니다...
            for query in sql_list:
                cursor.execute(query) # 쿼리 실행
                print(f"Query executed: {query}") # 각 쿼리 실행 후, 실행된 쿼리를 출력
            connection.commit()
            return True

In [54]:

sql = '''
ALTER TABLE jibun_rnaddrkor ADD PNU VARCHAR(19); 
UPDATE jibun_rnaddrkor SET PNU = CONCAT(`법정동코드`, `산여부`, LPAD(`지번본번(번지)`, 4, '0'), LPAD(`지번부번(호)`, 4, '0'));
'''
multi_query_update(sql)

Query executed: ALTER TABLE jibun_rnaddrkor ADD PNU VARCHAR(19);
Query executed: UPDATE jibun_rnaddrkor SET PNU = CONCAT(`법정동코드`, `산여부`, LPAD(`지번본번(번지)`, 4, '0'), LPAD(`지번부번(호)`, 4, '0'));


True

In [55]:
sql = "SELECT PNU FROM jibun_rnaddrkor LIMIT 5;"
query_get(sql)

[{'PNU': '1111010100001300003'},
 {'PNU': '1111010100001290002'},
 {'PNU': '1111010100001290003'},
 {'PNU': '1111010100001310001'},
 {'PNU': '1111010100001290001'}]

In [56]:
sql = '''
SELECT A.`도로명주소관리번호`, A.`도로명주소1`, A.`도로명주소2`, B.PNU 
FROM full_rna_addr A 
LEFT JOIN jibun_rnaddrkor B 
ON A.`도로명주소관리번호` = B.`도로명주소관리번호` 
LIMIT 10;
'''
query_get(sql)

[{'도로명주소관리번호': '11110101310001200009400000',
  '도로명주소1': '서울특별시 종로구 자하문로 94 (청운동)',
  '도로명주소2': '서울특별시 종로구 자하문로 94',
  'PNU': None},
 {'도로명주소관리번호': '11110101310001200009600000',
  '도로명주소1': '서울특별시 종로구 자하문로 96 (청운동, 평안빌)',
  '도로명주소2': '서울특별시 종로구 자하문로 96',
  'PNU': None},
 {'도로명주소관리번호': '11110101310001200009800000',
  '도로명주소1': '서울특별시 종로구 자하문로 98 (청운동, 청운빌라)',
  '도로명주소2': '서울특별시 종로구 자하문로 98',
  'PNU': None},
 {'도로명주소관리번호': '11110101310001200009900003',
  '도로명주소1': '서울특별시 종로구 자하문로 99-3 (청운동, 풍림팍사이드빌라)',
  '도로명주소2': '서울특별시 종로구 자하문로 99-3',
  'PNU': None},
 {'도로명주소관리번호': '11110101310001200009900004',
  '도로명주소1': '서울특별시 종로구 자하문로 99-4 (청운동)',
  '도로명주소2': '서울특별시 종로구 자하문로 99-4',
  'PNU': '1111010100001300003'},
 {'도로명주소관리번호': '11110101310001200010000000',
  '도로명주소1': '서울특별시 종로구 자하문로 100 (청운동)',
  '도로명주소2': '서울특별시 종로구 자하문로 100',
  'PNU': None},
 {'도로명주소관리번호': '11110101310001200010000001',
  '도로명주소1': '서울특별시 종로구 자하문로 100-1 (청운동)',
  '도로명주소2': '서울특별시 종로구 자하문로 100-1',
  'PNU': None},
 {'도로명주소관리번호': 

In [57]:
from sqlalchemy import create_engine
import pandas as pd

sql = '''
SELECT A.`도로명주소관리번호`, COUNT(DISTINCT B.PNU) AS PNU_COUNT
FROM full_rna_addr A
LEFT JOIN jibun_rnaddrkor B ON A.`도로명주소관리번호` = B.`도로명주소관리번호`
GROUP BY A.`도로명주소관리번호`
ORDER BY PNU_COUNT DESC
LIMIT 10;
'''

engine = create_engine("mysql+pymysql://root:root@localhost:3306/address?charset=utf8", echo=True)

conn = engine.connect()
df1 = pd.read_sql(sql, con=conn)
conn.close()

df1

2024-05-24 02:08:06,666 INFO sqlalchemy.engine.Engine SELECT DATABASE()
2024-05-24 02:08:06,668 INFO sqlalchemy.engine.Engine [raw sql] {}
2024-05-24 02:08:06,670 INFO sqlalchemy.engine.Engine SELECT @@sql_mode
2024-05-24 02:08:06,670 INFO sqlalchemy.engine.Engine [raw sql] {}
2024-05-24 02:08:06,672 INFO sqlalchemy.engine.Engine SELECT @@lower_case_table_names
2024-05-24 02:08:06,672 INFO sqlalchemy.engine.Engine [raw sql] {}
2024-05-24 02:08:06,675 INFO sqlalchemy.engine.Engine BEGIN (implicit)
2024-05-24 02:08:06,676 INFO sqlalchemy.engine.Engine DESCRIBE `address`.`
SELECT A.``도로명주소관리번호``, COUNT(DISTINCT B.PNU) AS PNU_COUNT
FROM full_rna_addr A
LEFT JOIN jibun_rnaddrkor B ON A.``도로명주소관리번호`` = B.``도로명주소관리번호``
GROUP BY A.``도로명주소관리번호``
ORDER BY PNU_COUNT DESC
LIMIT 10;
`
2024-05-24 02:08:06,676 INFO sqlalchemy.engine.Engine [raw sql] {}
2024-05-24 02:08:06,679 INFO sqlalchemy.engine.Engine 
SELECT A.`도로명주소관리번호`, COUNT(DISTINCT B.PNU) AS PNU_COUNT
FROM full_rna_addr A
LEFT JOIN jibun_r

도로명주소관리번호  PNU_COUNT
0  27170109300701600003300000        637
1  27170103422922100003000000        481
2  11140153310002100024600000        448
3  52710330301600100150000000        417
4  27170103314301200044600000        417
5  30200133316706000022000001        410
6  52710330301600200018100000        409
7  27170103314300400023000000        408
8  48840390334304500154500000        384
9  48250119333505100050200000        379

In [58]:
rna_addr_list = df1['도로명주소관리번호'].tolist()

sql = f'''
SELECT * FROM full_rna_addr WHERE `도로명주소관리번호` IN {str(rna_addr_list).replace('[','(').replace(']',')')}
'''

engine = create_engine("mysql+pymysql://root:root@localhost:3306/address?charset=utf8", echo=True)

conn = engine.connect()
df2 = pd.read_sql(sql, con=conn)
conn.close()
df2

2024-05-24 02:09:51,165 INFO sqlalchemy.engine.Engine SELECT DATABASE()
2024-05-24 02:09:51,166 INFO sqlalchemy.engine.Engine [raw sql] {}
2024-05-24 02:09:51,169 INFO sqlalchemy.engine.Engine SELECT @@sql_mode
2024-05-24 02:09:51,169 INFO sqlalchemy.engine.Engine [raw sql] {}
2024-05-24 02:09:51,170 INFO sqlalchemy.engine.Engine SELECT @@lower_case_table_names
2024-05-24 02:09:51,170 INFO sqlalchemy.engine.Engine [raw sql] {}
2024-05-24 02:09:51,173 INFO sqlalchemy.engine.Engine BEGIN (implicit)
2024-05-24 02:09:51,173 INFO sqlalchemy.engine.Engine DESCRIBE `address`.`
SELECT * FROM full_rna_addr WHERE ``도로명주소관리번호`` IN ('27170109300701600003300000', '27170103422922100003000000', '11140153310002100024600000', '52710330301600100150000000', '27170103314301200044600000', '30200133316706000022000001', '52710330301600200018100000', '27170103314300400023000000', '48840390334304500154500000', '48250119333505100050200000')
`
2024-05-24 02:09:51,174 INFO sqlalchemy.engine.Engine [raw sql] {}
20

새우편번호                                 도로명주소1                      도로명주소2  \
0  04563                서울특별시 중구 청계천로 246 (방산동)           서울특별시 중구 청계천로 246   
1  41774    대구광역시 서구 문화로 230 (평리동, 서대구역반도유보라센텀)            대구광역시 서구 문화로 230   
2  41763       대구광역시 서구 당산로 446 (평리동, 서대구영무예다음)            대구광역시 서구 당산로 446   
3  41767  대구광역시 서구 서대구로29길 30 (평리동, 서대구역화성파크드림)         대구광역시 서구 서대구로29길 30   
4  41716      대구광역시 서구 고성로 33 (원대동3가, 서대구센트럴자이)             대구광역시 서구 고성로 33   
5  34059              대전광역시 유성구 자운로 220-1 (추목동)         대전광역시 유성구 자운로 220-1   
6  50811                 경상남도 김해시 인제로 502 (삼방동)            경상남도 김해시 인제로 502   
7  52455                  경상남도 남해군 창선면 흥선로 1545       경상남도 남해군 창선면 흥선로 1545   
8  55365             전북특별자치도 완주군 이서면 콩쥐팥쥐로 1500  전북특별자치도 완주군 이서면 콩쥐팥쥐로 1500   
9  55365                전북특별자치도 완주군 이서면 혁신로 181     전북특별자치도 완주군 이서면 혁신로 181   

                    도로명주소관리번호      시도명  
0  11140153310002100024600000    서울특별시  
1  27170103314300400023000000    대구광역시  
2  27170103314301200044600000    대구광역시  
3  27170103422922100003000000    대구광역시  
4  27170109300701600003300000    대구광역시  
5  30200133316706000022000001    대전광역시  
6  48250119333505100050200000     경상남도  
7  48840390334304500154500000     경상남도  
8  52710330301600100150000000  전북특별자치도  
9  52710330301600200018100000  전북특별자치도

In [59]:
merged = df1.merge(df2, on='도로명주소관리번호').sort_values('PNU_COUNT', ascending=False)
merged

도로명주소관리번호  PNU_COUNT  새우편번호  \
0  27170109300701600003300000        637  41716   
1  27170103422922100003000000        481  41767   
2  11140153310002100024600000        448  04563   
3  52710330301600100150000000        417  55365   
4  27170103314301200044600000        417  41763   
5  30200133316706000022000001        410  34059   
6  52710330301600200018100000        409  55365   
7  27170103314300400023000000        408  41774   
8  48840390334304500154500000        384  52455   
9  48250119333505100050200000        379  50811   

                                  도로명주소1                      도로명주소2      시도명  
0      대구광역시 서구 고성로 33 (원대동3가, 서대구센트럴자이)             대구광역시 서구 고성로 33    대구광역시  
1  대구광역시 서구 서대구로29길 30 (평리동, 서대구역화성파크드림)         대구광역시 서구 서대구로29길 30    대구광역시  
2                서울특별시 중구 청계천로 246 (방산동)           서울특별시 중구 청계천로 246    서울특별시  
3             전북특별자치도 완주군 이서면 콩쥐팥쥐로 1500  전북특별자치도 완주군 이서면 콩쥐팥쥐로 1500  전북특별자치도  
4       대구광역시 서구 당산로 446 (평리동, 서대구영무예다음)            대구광역시 서구 당산로 446    대구광역시  
5              대전광역시 유성구 자운로 220-1 (추목동)         대전광역시 유성구 자운로 220-1    대전광역시  
6                전북특별자치도 완주군 이서면 혁신로 181     전북특별자치도 완주군 이서면 혁신로 181  전북특별자치도  
7    대구광역시 서구 문화로 230 (평리동, 서대구역반도유보라센텀)            대구광역시 서구 문화로 230    대구광역시  
8                  경상남도 남해군 창선면 흥선로 1545       경상남도 남해군 창선면 흥선로 1545     경상남도  
9                 경상남도 김해시 인제로 502 (삼방동)            경상남도 김해시 인제로 502     경상남도

서대구센트럴자이, 서대구역화성파크드림 등이 지번이 많다   
서울특별시 중구 청계천로 246 (방산동) - 방산시장, 시장의 가게마다 지번이 부여되어 지번이 많다고 한다